src/hooks/useUser.jsx

```diff
@@ -0,0 +1,52 @@
+import { useContext, useCallback, useState } from 'react'
+import Context from '../contexts/UserContext'
+import loginService from './../services/login'
+import logoutService from './../services/logout'
+
+export default function useUser () {
+  const {token, setToken} = useContext(Context)
```

Comment:
here you could use what i suggested above 
```suggestion
  const {token, setToken} = useUserContext()
```